purview/api/v1.py

```diff
@@ -103,4 +122,110 @@ def get_resource_story(resource, uid):
 
     results['meta']['related_nodes'].update(get_corelated_nodes(results_unordered))
 
+    # Adding the artifact itself if it's story is not available
+    if len(results['data']) == 0:
+        results['data'].append(item.serialized_all)
+        results['data'][0]['resource_type'] = item.__label__
+
     return jsonify(results)
+
+
+@api_v1.route('/allstories/<resource>/<uid>')
+def get_resource_all_stories(resource, uid):
+    """
+    Get all unique stories of an artifact from Neo4j.
+
+    :param str resource: a resource name that maps to a neomodel class
+    :param str uid: the value of the UniqueIdProperty to query with
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raises NotFound: if the item is not found
+    :raises ValidationError: if an invalid resource was requested
+    """
+    item = get_neo4j_node(resource, uid)
+    if not item:
+        raise NotFound('This item does not exist')
+
+    uid_name = item.unique_id_property
+    forward_query = create_story_query(item, uid_name, uid)
+    backward_query = create_story_query(item, uid_name, uid, reverse=True)
+
+    def _get_partial_stories(query, resources_to_expand=None):
+        if resources_to_expand is None:
+            resources_to_expand = []
+
+        results_list = []
+        results, _ = db.cypher_query(query)
+
+        if not results:
+            return [results_list]
+
+        # Creating a list of lists where each list is a collection of node ids
+        # of the nodes present in that particular story path.
+        # Paths are re-sorted in the ascending order for simplifying the logic below
+        path_nodes_id = []
+        for path in reversed(results):
+            path_nodes_id.append([node.id for node in path[0].nodes])
+
+        unique_paths = []
+        for index, node_set in enumerate(path_nodes_id[:-1]):
+            unique = True
+            for alternate_set in path_nodes_id[index + 1:]:
+                # If the node_set is a subset of alternate_set,
```

Comment:
Optional: You could take some of the words from the line below and put it at the end of this line. It's a waste of space have the line break so early.